src/directives/formly-field.js

```diff
@@ -87,9 +87,9 @@ function formlyField($http, $q, $compile, $templateCache, $interpolate, formlyCo
       if (!key || !model) {
         return
       }
+
       // If we are working with a number then $parse wont work, default back to the old way for now
-      if (angular.isNumber(key)) {
-        // TODO: Fix this so we can get several levels instead of just one with properties that are numeric
+      if (angular.isNumber(key) || (/\d/.test(key) && key.indexOf('[') === -1)) {
```

Comment:
Hmmm, I believe that my example demonstrated the issue pretty well. I guess, here's a test. We need to make this pass :-)

```javascript
it('should work with nested keys with numbers in the key', () => {
  const key = 'foo3bar.baz4foobar'
  const defaultValue = 'baz'
  scope.fields = [
    {template: input, key, defaultValue}
  ]
  scope.model = {}

  compileAndDigest()
  expect(scope.model.foo3bar.baz4foobar).to.equal(defaultValue)
})
```

Does that make sense? I think what we have now would actually cause this test to fail.